### PR TITLE
chore: add code snippet for swift api query with json objects

### PR DESCRIPTION
**Pull request by phantumcode**
#### Description of changes:
* Adding additional code snippet for custom query inputs with JSON object in API category advanced workflow for Swift platform.

#### Related GitHub issue #, if available:
https://github.com/aws-amplify/amplify-swift/issues/3471

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [x] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 1]
```diff
@@ -175,3 +175,31 @@ import android9 from '/src/fragments/lib/graphqlapi/android/advanced-workflows/5
...
+
+<InlineFilter filters={['swift']}>
+
+## Custom Request Inputs as JSON Objects
+
+If you have custom query inputs defined, you can convert the input objects to Json objects as part of the requests.
+
+```swift
+extension GraphQLRequest {
+    static func get(filter: CustomRequestFilter) -> GraphQLRequest<JSONValue> {
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:187]: I think some of this is ambiguous. We should start off with a schema example that represents the use case, "custom query inputs". Then define the CustomRequestFilter type, looking at the code below, it's at least Encodable. When the developer brings the code over to their IDE it should compile and run

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 2]
```diff
@@ -175,3 +175,31 @@ import android9 from '/src/fragments/lib/graphqlapi/android/advanced-workflows/5
...
+        let document = """
+        ....
+        """
+        
+        let data = try? JSONEncoder().encode(filter)
+        let jsonObject = try? JSONSerialization.jsonObject(with: data!, options: [])
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:193]: can we avoid force casting?

**Review by lawmicha — CHANGES_REQUESTED** [review]

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx (hunk 1 above)
Comment (phantumcode) [reply src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:187]: I've updated the code sample based on your feedback

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx (hunk 2 above)
Comment (phantumcode) [reply src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:193]: Removed force casting and add guard/throw check

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 3]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
 
 <Fragments fragments={{ flutter: flutter3 }} />
 
+<InlineFilter filters={['swift']}>
+
+## Custom Request Inputs
+
+The **GraphQLRequest**'s variables property takes in a JSON object. You can define this as an Encodable type and serialize it to a JSON object for the request.
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:97]:
```suggestion
The `GraphQLRequest`'s `variables` property takes in a JSON object. You can define this as an **Encodable** type and serialize it to a JSON object for the request.
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 4]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+
+Define the type that matches the expected GraphQL input type:
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:99]:
```suggestion
Define the type that matches the expected input type from your GraphQL service.
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 5]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+
+```swift
+struct CreateTodoInput: Encodable {
+  let id: String!
+  let name: String!
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:104]:
```suggestion
  let id: String
  let name: String
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 6]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+  let description: String?
+}
+```
+
+Then create the GraphQLRequest
+```swift
+extension GraphQLRequest {
+    static func createToDo(input: CreateTodoInput) throws -> GraphQLRequest<TODO> {
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx]:
```suggestion
    static func createTodo(input: CreateTodoInput) throws -> GraphQLRequest<Todo> {
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 7]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+      let operationName = "createTodo"
+      let document = """
+      query createTodo($id: ID!) {
+        \(operationName)(id: $id) {
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:116]:
```suggestion
      query createTodo($input: CreateTodoInput!) {
        \(operationName)(input: $input) {
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 8]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+  let id: String!
+  let name: String!
+  let description: String?
+}
+```
+
+Then create the GraphQLRequest
+```swift
+extension GraphQLRequest {
+    static func createToDo(input: CreateTodoInput) throws -> GraphQLRequest<Todo> {
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:112]:
```suggestion
    static func createTodo(input: CreateTodoInput) throws -> GraphQLRequest<Todo> {
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 9]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+struct CreateTodoInput: Encodable {
+  let id: String!
+  let name: String!
+  let description: String?
+}
+```
+
+Then create the GraphQLRequest
+```swift
+extension GraphQLRequest {
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:111]:
```suggestion
```swift
import Foundation

extension GraphQLRequest {
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 10]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+
+```swift
+struct CreateTodoInput: Encodable {
+  let id: String!
+  let name: String!
+  let description: String?
+}
+```
+
+Then create the GraphQLRequest
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:109]:
```suggestion
Then create the `GraphQLRequest`
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 11]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+          document: document,
+          variables: ["input": jsonObject],
+          responseType: Todo.self,
+          decodePath: operationName
+      )
+    }
+}
+```
+
+</InlineFilter>
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:137]:
```suggestion
```

The `input` object is encoded as a data object and then serialized into the JSON object.

</InlineFilter>
```

File: src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx [hunk 12]
```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
...
+      return GraphQLRequest<Todo>(
+          document: document,
+          variables: ["input": jsonObject],
+          responseType: Todo.self,
+          decodePath: operationName
+      )
+    }
+}
+```
+
```
Comment (lawmicha) [inline src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx:136]:
```suggestion

The input object is encoded as a data object and then serialized into the JSON object.


```

**Review by lawmicha — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**